### DISCUSSION TITLE
Friday night funkin kade engine 1.6.1 How to add an event animation like ugh

### DISCUSSION THREAD

**Discussion by Reyororo** [KadeArchive/Kade-Engine#1653] [Q&A] [answered]
How do I add an event animation like ugh?

**Comment by napstaa967** [comment, marked as answer]
i won't go into it but ugh uses an alt anim for up and i don't know how that works so maybe wait until source is out

if you want to add an animation that plays after a specific time inside a song, like gf cheering in blammed you can go to playstate.hx and go behind the curstage switch then add a line like this:

modify the conditions to whatever you want to do
```hx
if (curBeat % 16 == 15 && SONG.song == 'Tutorial' && dad.curCharacter == 'gf' && curBeat > 16 && curBeat < 48)
{
			boyfriend.playAnim('hey', true);
			dad.playAnim('cheer', true);
}

**Reply by napstaa967** [reply]
% is divide, so the conditions are:
if curbeat divided by 16 is 15
and song is tutorial
and curcharacter is gf
and the beat is bigger than 16
and the beat is smaller than 48

**Reply by isakube** [reply]
> % is divide, so the conditions are:

yeah okay
% is something called modulus and it literally saved my life the first time i found it lol
basically you take the remainder of a division so you're not that far off actually

if i come off as rude, fuck it it's early here and im tired

**Reply by Reyororo** [reply]
> i won't go into it but ugh uses an alt anim for up and i don't know how that works so maybe wait until source is out
> 
> if you want to add an animation that plays after a specific time inside a song, like gf cheering in blammed you can go to playstate.hx and go behind the curstage switch then add a line like this:
> 
> modify the conditions to whatever you want to do
> 
> ```hx
> if (curBeat % 16 == 15 && SONG.song == 'Tutorial' && dad.curCharacter == 'gf' && curBeat > 16 && curBeat < 48)
> {
> 			boyfriend.playAnim('hey', true);
> 			dad.playAnim('cheer', true);
> }
> ```

Thank you

**Reply by perngjiafu** [reply]
alt animations is in character.hx. if you wanna add a alt animation like ugh, you must enter in 

"animation.addbyPrefix('singUP-alt', 'ugh', 24, false);"
and add "ugh" in the .xml file for the sprite you want

to activate it, go to fnf debug mode, go to section data, and you'll see alternative animation with a check box.

thank you
